llmnrd.c

```diff
@@ -150,6 +150,24 @@ static void hostname_change_handle(char *hostname, size_t maxlen)
 	free(newname);
 }
 
+static bool write_pid_file(void)
+{
+	FILE *fp;
+
+	fp = fopen(PIDFILE, "w");
+	if ( ! fp ) {
+		log_err("Failed to open pid file %s: %s", PIDFILE, strerror(errno));
+		return false;
+	}
+
+	if (fprintf(fp, "%ji\n", (intmax_t) getpid()) < 0)
+		log_err("Failed to write pid to " PIDFILE);
```

Comment:
Nit: use a format specifier here:
```suggestion
		log_err("Failed to write pid to %s", PIDFILE);
```

Reply:
OK, I'll change it if you prefer that form, or is there a shortcoming other than style that I am missing?

Reply:
That's just personal style preference and I try to stick to it in general to avoid injecting format specifier through user controlled strings (https://owasp.org/www-community/attacks/Format_string_attack). This isn't really the case here given `PIDFILE` can only be set at compile time, but IMO it's still nice to stick to the pattern consistently, to avoid potential vulnerabilities if code is refactored, copied, etc.

Reply:
Thanks for explaining, and indeed it is safer if for some reason one sets PIDFILE to have a format specifier at compile time, like `/run/funky-%dir/pid`